modules/transports/core/vfs/src/main/java/org/apache/synapse/transport/vfs/VFSTransportSender.java

```diff
@@ -119,6 +125,44 @@ public void sendMessage(MessageContext msgCtx, String targetAddress,
             vfsOutInfo = (VFSOutTransportInfo) outTransportInfo;
         }
 
+        WriteLockObject lockObject = null;
+        String baseUri = null;
+        if (vfsOutInfo != null) {
+            if (vfsOutInfo.getSendFileSynchronously()) {
+                baseUri = getBaseUri(targetAddress);
+                if (baseUri != null) {
+                    if (!lockingObjects.containsKey(baseUri)) {
+                        lockingObjects.putIfAbsent(baseUri, new WriteLockObject());
+                        log.debug("New locking object created for Synchronous write|MapSize:" + lockingObjects.size());
```

Comment:
As a best practice, wrap your debug logs with guard statements in order to avoid the execution of potentially expensive operations like string concatinations, etc even when logging is not enabled.